clients/vault/src/cancellation.rs

```diff
@@ -168,18 +168,9 @@ impl<P: IssuePallet + ReplacePallet + Send + Sync> Canceller<P> for ReplaceCance
 
 // verbose drain_filter
 fn drain_expired(requests: &mut Vec<ActiveRequest>, current_height: u32) -> Vec<ActiveRequest> {
-	let mut expired = Vec::new();
-	let has_expired = |request: &ActiveRequest| current_height > request.parachain_deadline_height;
-	let mut i = 0;
-	while i != requests.len() {
-		if has_expired(&requests[i]) {
-			let req = requests.remove(i);
-			expired.push(req);
-		} else {
-			i += 1;
-		}
-	}
-	expired
+
+	requests.drain(..).filter(|active_req| current_height > active_req.parachain_deadline_height).collect()
```

Comment:
I think we have to be very careful here.
As the [docs](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.drain) say: Removes the specified range from the vector in bulk, returning all removed elements as an iterator. If the iterator is dropped before being fully consumed, it drops the remaining removed elements.

This means that the passed `requests` vector will be empty after calling this function. Whereas in the previous implementation, elements were only 'moved' from `requests` to `expired` if needed, keeping the original `requests` vector mostly intact.

Reply:
Indeed!, I just commited a new impl for this, so that the functionality is the same as before. I think this is what may have caused the error in Integration test (still not 100% sure).